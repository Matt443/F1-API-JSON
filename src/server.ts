import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}