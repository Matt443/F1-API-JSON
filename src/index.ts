import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
