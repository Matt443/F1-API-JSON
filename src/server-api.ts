import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
