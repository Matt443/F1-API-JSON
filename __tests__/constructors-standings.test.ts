import { getConstructorStandings } from "../src/server";

test("constrcutor standings", async () => {
    expect(await getConstructorStandings()).toMatchSnapshot();
});
