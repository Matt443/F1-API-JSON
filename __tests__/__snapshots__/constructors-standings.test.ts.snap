// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`constrcutor standings 1`] = `
[
  {
    "points": 656,
    "position": 1,
    "team": "Red Bull Racing RBPT",
  },
  {
    "points": 469,
    "position": 2,
    "team": "Ferrari",
  },
  {
    "points": 416,
    "position": 3,
    "team": "Mercedes",
  },
  {
    "points": 144,
    "position": 4,
    "team": "Alpine Renault",
  },
  {
    "points": 138,
    "position": 5,
    "team": "McLaren Mercedes",
  },
  {
    "points": 52,
    "position": 6,
    "team": "Alfa Romeo Ferrari",
  },
  {
    "points": 51,
    "position": 7,
    "team": "Aston Martin Aramco Mercedes",
  },
  {
    "points": 38,
    "position": 8,
    "team": "Haas Ferrari",
  },
  {
    "points": 36,
    "position": 9,
    "team": "AlphaTauri RBPT",
  },
  {
    "points": 8,
    "position": 10,
    "team": "Williams Mercedes",
  },
]
`;
